language/move-core/types/src/language_storage.rs

```diff
@@ -79,6 +103,145 @@ impl<'a> From<&'a TypeTag> for SafeTypeTag<'a> {
     }
 }
 
+struct TypeTagVisitor(u8);
```

Comment:
Maybe should comment that u8 is `depth` or make a struct with `u8` named as depth